src/Components/Components/src/PublicAPI.Unshipped.txt

```diff
@@ -3,4 +3,18 @@ Microsoft.AspNetCore.Components.ComponentBase.DispatchExceptionAsync(System.Exce
 Microsoft.AspNetCore.Components.RenderHandle.DispatchExceptionAsync(System.Exception! exception) -> System.Threading.Tasks.Task!
 *REMOVED*Microsoft.AspNetCore.Components.NavigationManager.ToAbsoluteUri(string! relativeUri) -> System.Uri!
 Microsoft.AspNetCore.Components.NavigationManager.ToAbsoluteUri(string? relativeUri) -> System.Uri!
+Microsoft.AspNetCore.Components.Sections.SectionContent
+Microsoft.AspNetCore.Components.Sections.SectionContent.ChildContent.get -> Microsoft.AspNetCore.Components.RenderFragment?
+Microsoft.AspNetCore.Components.Sections.SectionContent.ChildContent.set -> void
+Microsoft.AspNetCore.Components.Sections.SectionContent.Dispose() -> void
+Microsoft.AspNetCore.Components.Sections.SectionContent.IsDefaultContent.get -> bool
+Microsoft.AspNetCore.Components.Sections.SectionContent.IsDefaultContent.set -> void
```

Comment:
Do you think it would be viable to remove `IsDefaultContent` from the public API? We added it in https://github.com/dotnet/aspnetcore/pull/35982 as a workaround for a bug at that time, but we didn't worry too much about the usability of this flag since it was only used internally. With `<SectionContent>` becoming public, we probably wouldn't want to expose this flag since it will be really hard for people to make sense of.

I'm not actually sure if `IsDefaultContent` is still required now that we fixed the issues about the order in which prerendering occurs. The following comment from @MackinnonBuck may no longer apply, but I haven't checked:

> An alternative I considered was to always render the SectionContent for the default title, but only render the <title> element when the default title is actually determined. This sometimes works, but it breaks in cases like prerendering because <PageTitle> elements in the "app" root component still get rendered first.

If it's possible to remove `IsDefaultContent` completely (rather than just making it `internal`) that would be the best outcome but I'm not 100% sure if that's possible! Is that something you'd be able to check, @surayya-MS?

Reply:
Sure, we could make `IDefaultContent` `internal`. I cannot answer now if we could remove it completely. It is used in [HeadOutlet](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Web/src/Head/HeadOutlet.cs#L48). I would have to investigate that.

>I'm not actually sure if IsDefaultContent is still required now that we fixed the issues about the order in which prerendering occurs.

@SteveSandersonMS could you please point out the PRs where we fixed the issues about the order in which prerendering occurs?

@MackinnonBuck , do you think it is possible to remove `ISDefaultContent` completely?

Reply:
Yes, the rendering order issues were fixed in https://github.com/dotnet/aspnetcore/pull/40512, which was after Mackinnon implemented the `<head>` support in https://github.com/dotnet/aspnetcore/pull/34503.

Reply:
I removed `IsDefaultContent` and tried out the bug described in the PR https://github.com/dotnet/aspnetcore/pull/35982 . It is no longer an issue. 
Btw it wouldn't be possible to make `IsDefaultContent` `internal` because it has `Parameter` attribute.

Reply:
> It is no longer an issue

That's great news!

> It wouldn't be possible to make IsDefaultContent internal because it has Parameter attribute.

There might still be ways to achieve it (by hand-writing the logic that applies parameters in `SetParametersAsync` instead of auto-assigning them based on `[Parameter]` attributes) but if we can simply remove it that's a much better outcome.

Reply:
> There might still be ways to achieve it (by hand-writing the logic that applies parameters in SectionContent's SetParametersAsync instead of auto-assigning them based on [Parameter] attributes)

The challenge was to then apply it to RenderTreeBuilder here https://github.com/dotnet/aspnetcore/blob/main/src/Components/Web/src/Head/HeadOutlet.cs#L48. Couldn't figure out how to do it.

It a good thing we can remove it altogether.